src/cryptonote_core/tx_pool.h

```diff
@@ -130,23 +130,74 @@ namespace cryptonote
+     * in with an `.approved()` status.
      *
      * @param blink - a shared_ptr to the blink details
      * @param tvc - the verification results
      * @param blink_exists - will be set to true if the addition fails because the blink tx already
      * exists
      *
-     * @return true if the tx passes validations and has been added to tx/blink pools
+     * @return true if the tx passes validations and has been added to the tx pool.
+     */
+    bool add_new_blink(const std::shared_ptr<blink_tx> &blink, tx_verification_context& tvc, bool &blink_exists);
+
+    /**
+     * @brief attempts to add blink transaction information about an existing blink transaction
+     *
+     * This method takes an approved blink_tx and records it in the known blinks data.  No check is
+     * done that the transaction actually exists on the blockchain or mempool.  It is assumed that
+     * the given shared_ptr is a new blink that is not yet shared between threads (and thus doesn't
+     * need locking): sharing is expected only after it is added to the blinks via this method.
+     *
+     * NB: this function assumes that the given blink tx is valid and approved but does *not* check
+     * it (except when compiling in debug mode).
+     *
+     * @param blink the blink_tx shared_ptr
+     * @param have_lock can be specified as true to avoid taking out a unique lock on the blinks
+     * data structure; it should only be specified if a unique lock on the blink data is already
+     * held externally, i.e. by obtaining a lock with `blink_unique_lock`.
+     *
+     * @return true if the blink data was recorded, false if the given tx was already known.
+     */
+    bool add_existing_blink(std::shared_ptr<blink_tx> blink, bool have_lock = false);
+
+    /**
+     * @brief accesses blink tx details if the given tx hash is a known, approved blink tx, nullptr
+     * otherwise.
+     *
+     * @param tx_hash the hash of the tx to access
+     * @param have_lock can be specified as true to avoid taking out a shared lock; it should only
+     * be specified if a shared lock on the blink data is already held externally.
+     */
+    std::shared_ptr<blink_tx> get_blink(const crypto::hash &tx_hash, bool have_lock = false) const;
+
+    /**
+     * Equivalent to `(bool) get_blink(...)`, but slightly more efficient when the blink information
+     * isn't actually needed beyond an existance test (as it avoids copying the shared_ptr).
      */
-    bool add_blink(const std::shared_ptr<blink_tx> &blink, tx_verification_context& tvc, bool &blink_exists);
+    bool has_blink(const crypto::hash &tx_hash, bool have_lock = false) const;
 
     /**
-     * @brief accesses blink tx details if the given tx hash is a known blink tx, nullptr otherwise.
+     * @brief takes a map of blink { height -> [tx_hashes] } and records any that we don't already
+     * know about (and are not before the immutable height) in the internal "m_want_blinks" to
```

Comment:
`"m_missing_blinks"`

Reply:
This method (and m_missing_blinks) got removed in PR #927.